src/FsAutoComplete/LspServers/AdaptiveServerState.fs

```diff
@@ -307,10 +372,12 @@ type AdaptiveState(lspClient: FSharpLspClient, sourceTextFactory: ISourceTextFac
     let checkUnusedOpens =
       async {
         try
-          use progress = new ServerProgressReport(lspClient)
+          use progress = progressLookup.CreateProgressReport(lspClient, cancellable = true)
           do! progress.Begin($"Checking unused opens {fileName}...", message = filePathUntag)
 
-          let! unused = UnusedOpens.getUnusedOpens (tyRes.GetCheckResults, getSourceLine)
+          let! unused =
+            UnusedOpens.getUnusedOpens (tyRes.GetCheckResults, getSourceLine)
+            |> Async.withCancellation progress.CancellationToken
```

Comment:
`ServerProgressReport` now has a `CancellationToken` we can use to cancel operations.